hypervisor/boot/cmdline.c

```diff
@@ -38,16 +38,16 @@ int32_t parse_hv_cmdline(void)
 	do {
 		while (*start == ' ')
 			start++;
-
-		end = start + 1;
-		while ((*end != ' ') && ((*end) != '\0'))
-			end++;
-
-		if (!handle_dbg_cmd(start, (int32_t)(end - start))) {
-			/* if not handled by handle_dbg_cmd, it can be handled further */
+		if ((*start) != '\0') {
+			end = start + 1;
+			while ((*end != ' ') && ((*end) != '\0'))
+				end++;
+
+			if (!handle_dbg_cmd(start, (int32_t)(end - start))) {
+				/* if not handled by handle_dbg_cmd, it can be handled further */
+			}
+			start = end + 1;
```

Comment:
if we came here because *end == '\0' in line 43,  start may point not-exist space

Reply:
Yes it is, I will fix it as well, Thank you for point it out.